tinycrypt/ecc.c

```diff
@@ -289,17 +289,29 @@ uECC_word_t uECC_vli_equal(const uECC_word_t *left, const uECC_word_t *right)
 	uECC_word_t tmp1, tmp2;
 	volatile int i;
 
-	for (i = NUM_ECC_WORDS - 1; i >= 0; --i) {
+	/* Start from a random location and check the correct number of iterations */
+	int flow_monitor = 0;
+	int start_offset = 1 + mbedtls_platform_random_in_range(NUM_ECC_WORDS - 1);
+
+	for (i = start_offset; i < NUM_ECC_WORDS; ++i) {
+		tmp1 = left[i];
+		tmp2 = right[i];
+		flow_monitor++;
+		diff |= (tmp1 ^ tmp2);
+	}
+
+	for (i = 0; i < start_offset; ++i) {
 		tmp1 = left[i];
 		tmp2 = right[i];
+		flow_monitor++;
 		diff |= (tmp1 ^ tmp2);
 	}
 
-	/* i should be -1 now */
+	/* Random delay to increase security */
 	mbedtls_platform_random_delay();
-	diff |= i ^ -1;
 
-	return diff;
+	/* Return 0 only when diff is 0 and flow_counter is equal to NUM_ECC_WORDS */
+	return (diff | (flow_monitor ^ NUM_ECC_WORDS));
```

Comment:
Shouldn't there be a cast to "uECC_word_t" here?

Reply:
Yes, It should be. I changed the type of flow_monitor to"uECC_word_t"